storage/innobase/srv/srv0start.cc

```diff
@@ -3913,7 +3914,7 @@ static lsn_t srv_shutdown_log() {
 
   srv_wake_log_tracker_thread();
 
-  if (srv_downgrade_logs) {
+  if (srv_downgrade_logs || dd_init_failed_during_upgrade) {
     ut_a(!srv_read_only_mode);
 
     log_files_downgrade(*log_sys);
```

Comment:
when I implemented this, it was only for downgrades to 5.7

Which version of redo it would downgrade to? I think this needs to handle the redo version for 8.0

Reply:
Indeed, it writes 5.7 redo header number

  /* Write old version */
  mach_write_to_4(buf + LOG_HEADER_FORMAT, LOG_HEADER_FORMAT_5_7_9);